cmd/packer-sdc/internal/fix/gocty.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package fix
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"golang.org/x/mod/modfile"
+)
+
+const (
+	oldPath     string = "github.com/zclconf/go-cty"
+	newPath     string = "github.com/nywilken/go-cty"
+	newVersion  string = "1.12.1"
+	modFileGlob string = "go.mod"
+)
+
+var goctyFix = fix{
+	name:        "gocty",
+	description: "Adds a replace directive for github.com/zclconf/go-cty to github.com/nywilken/go-cty",
+	fileGlob:    modFileGlob,
+	fixer: GoCtyFix{
+		OldPath:    oldPath,
+		NewPath:    newPath,
+		NewVersion: newVersion,
+	},
+}
+
+type GoCtyFix struct {
+	OldPath, NewPath, NewVersion string
+}
+
+func (f GoCtyFix) modFileFormmatedVersion() string {
+	return fmt.Sprintf("v%s", f.NewVersion)
+}
+
+func (f GoCtyFix) isUnfixed(modFilePath string, data []byte) (bool, error) {
```

Comment:
Coming back on this function, I'm a bit confused when looking at the name since I interpret it as a fix that got removed, not a fix that needs to happen.
May I suggest calling it `needsFix`? That would make it clearer imho